widget/list.go

```diff
@@ -19,14 +19,18 @@ var _ fyne.Widget = (*List)(nil)
 type List struct {
 	BaseWidget
 
-	Length         func() int
-	CreateItem     func() fyne.CanvasObject
-	UpdateItem     func(index int, item fyne.CanvasObject)
-	OnItemSelected func(index int)
-	selectedItem   *listItem
-	selectedIndex  int
-	itemMin        fyne.Size
-	offsetY        int
+	Length           func() int
+	CreateItem       func() fyne.CanvasObject
+	UpdateItem       func(index int, item fyne.CanvasObject)
+	OnItemSelected   func(index int)
+	visibleItemCount int
+	firstItemIndex   int
```

Comment:
I think you can work without these additional fields cached.
If you calculate the offset position based on index then compare it to the scroll offsets, you'll know if it's visible.

Reply:
You're right - although those fields were already cached - I just move their location from the renderer to the widget.  If we already had that information seems to be more readable and simpler with less calcs.  No strong opinions either way though.

Reply:
I'll go with whatever another reviewer thinks. Am just not a fan of caching render details in the widget if it can be avoided.
Also I have a suspicion that using the approach of rounding to the nearest cell that scrolling an item that is off-bottom into the screen might bring it further into the view than necessary (as it's not checking the actual view dimensions when determining what to show).

Reply:
Tested (as mentioned below) and the maths works out fine, sorry for the noise.

Reply:
Crash fixed.

Reply:
Okay cached values moved back to renderer